src/android/LocalyticsPlugin.java

```diff
@@ -992,6 +1003,9 @@ public void run() {
         } else if (action.equals("setMessagingListener")) {
             messagingListener = new CDMessagingListener(callbackContext, inAppCampaignCache, pushCampaignCache, placesCampaignCache);
             Localytics.setMessagingListener(messagingListener);
+            PluginResult result = new PluginResult(PluginResult.Status.OK);
+            result.setKeepCallback(true);
+            callbackContext.sendPluginResult(result);
             return true;
         } else if (action.equals("removeMessagingListener")) {
             messagingListener = null;
```

Comment:
Don't we need a callbackContext.success() here for removeMessagingListener so that the callback that's kept is actually freed.

Reply:
nope - the `PluginResult(PluginResult.Status.OK)` is inherently a `success()`.  `success()` under the hood basically does this code (minus the keep callback)